ods_ci/tests/Tests/100__deploy/130__operators/140__serverless_operator/140__serverless_operator.robot

```diff
@@ -0,0 +1,111 @@
+*** Settings ***
+Documentation    Test Cases to verify Serverless installation
+Library         Collections
+Library         SeleniumLibrary
+Library         OpenShiftLibrary
+Resource        ../../../../Resources/Page/OCPDashboard/OCPDashboard.resource
+Suite Setup      Suite Setup
+Suite Teardown   Suite Teardown
+
+
+*** Variables ***
+${RHOAI_APPNAME}  Red Hat OpenShift AI
+${RHOAI_OPERATOR_NAME}    Red Hat OpenShift AI
+${KSERVE_APPNAME}  Red Hat OpenShift Serverless
+${KSERVE_OPERATOR_NAME}    Red Hat OpenShift Serverless
+${KNATIVESERVING_NS}    knative-serving
+${RHOAI_OPERATOR_NS}    redhat-ods-operator
+
+
+*** Test Cases ***
+Validate DSC creates all Serverless CRs
+    [Documentation]  The purpose of this Test Case is to validate the creation
+    ...    of KServe Custom Resources
+    [Tags]  Operator    ODS-2600
+    Check And Install Operator In Openshift    ${RHOAI_APPNAME}    ${RHOAI_OPERATOR_NAME}
+    Check And Install Operator In Openshift    ${KSERVE_APPNAME}    ${KSERVE_OPERATOR_NAME}
+    Is Resource Present     KnativeServing    knative-serving     ${KNATIVESERVING_NS}
+    Check Status     oc get KnativeServing knative-serving -n ${KNATIVESERVING_NS} -o json | jq '.status.conditions[] | select(.type=="Ready") | .status'     KnativeServing    "True"    # robocop: disable
+    Is Resource Present     Gateway    knative-ingress-gateway     ${KNATIVESERVING_NS}
+    Is Resource Present     Gateway    knative-local-gateway     ${KNATIVESERVING_NS}
+    Is Resource Present     Service    knative-local-gateway     istio-system
+    Is Resource Present     deployment    controller     ${KNATIVESERVING_NS}
+    Check Pods Number    ${KNATIVESERVING_NS}    app.kubernetes.io/component=controller
+    ${pod_names}=    Get Pod Names    ${KNATIVESERVING_NS}    app.kubernetes.io/component=controller
+    Verify Containers Have Zero Restarts    ${pod_names}    ${KNATIVESERVING_NS}
+    ${podname}=    Get Pod Name   ${RHOAI_OPERATOR_NS}    label_selector=name=rhods-operator
```

Comment:
this will only work on RHOAI build? how to handle ODH ?

Reply:
I need to figure out how this can be customized to upstream/downstream execution, maybe through tagging, but that would involve another Test Case...

Reply:
We can determine which product is under the test via `PRODUCT` variable in the `tests-variables.yml` file.